service/src/main/kotlin/fi/espoo/evaka/attachment/AttachmentQueries.kt

```diff
@@ -65,26 +56,36 @@ fun Database.Transaction.insertAttachment(
 fun Database.Read.getAttachment(id: AttachmentId): Attachment? = this
     .createQuery(
         """
-        SELECT id, name, content_type
+        SELECT id, name, content_type, uploaded_by_employee, uploaded_by_person, application_id, income_statement_id
         FROM attachment
         WHERE id = :id
         """
     )
     .bind("id", id)
-    .mapTo<Attachment>()
+    .map { row ->
```

Comment:
This could be simplified a bit using the query-specific row mapper technique I mentioned a couple of weeks ago :nerd_face: 

Since we only want to specialize handling of one field that has a specific type, something like this should work:

```kotlin
.registerColumnMapper(AttachmentParent::class.java) { row ->
  val applicationId = row.mapColumn<ApplicationId?>("application_id")
  val incomeStatementId = row.mapColumn<IncomeStatementId?>("income_statement_id")
  if (applicationId != null) AttachmentParent.Application(applicationId)
  else if (incomeStatementId != null) AttachmentParent.IncomeStatement(incomeStatementId)
  else AttachmentParent.None
}
.mapTo<Attachment>()
```

= "JDBI, if you encounter a need to map an AttachmentParent somewhere, use this piece of code to do it"

Reply:
Never mind, the API is a bit different than I thought, so it would make things a bit too complicated

Reply:
```kotlin
.registerRowMapper(AttachmentParent::class.java) { rs, ctx ->
  val applicationId = ctx.mapColumn<ApplicationId?>(rs, "application_id")
  val incomeStatementId = ctx.mapColumn<IncomeStatementId?>(rs, "income_statement_id")
  if (applicationId != null) AttachmentParent.Application(applicationId)
  else if (incomeStatementId != null) AttachmentParent.IncomeStatement(incomeStatementId)
  else AttachmentParent.None
}
```

Actually, it's probably not that bad :thinking: 

:shrug:

Reply:
Great suggestion, changed!